File: Gridland_metro/gridland_metro.c
//gridland metro
#include<stdio.h>
typedef long long ll;
int main()
{
    ll list1[3];
    ll i,j,m,n;
    int k;
    ll ans=0;
    ll temp[100000];
    ll r,c1,c2;
    i=0;
    j=0;
    scanf("%lld %lld %d",&m,&n,&k);
    while(j<k)
    {
        scanf("%lld %lld %lld",&r,&c1,&c2);
        temp[r]=temp[r]+c2-c1+1;
        ans+=temp[r];
        j++;
    }
    ans=(n*m)-ans;
    printf("%lld",ans);
    return 0;
}

File: Gridland_metro/gridland_metro.py
#gridland metro
ip=[int(x) for x in raw_input().split()]
temp={}
j=0
ans=0
while(j<ip[2]):
    ip1=[int(y) for y in raw_input().split()]
    r=ip1[0]
    c1=ip1[1]
    c2=ip1[2]
    if temp.has_key(r):
        temp[r].extend([c1,c2])
    else:
        temp[r]=[c1,c2]
    j+=1
print temp
##for key,value in temp.iteritems():
##    for a in range(len(value)):
##        frm=a
##        till=a+1
